### DISCUSSION TITLE
How to restore settings from previous installation

### DISCUSSION THREAD

**Discussion by mora-phi** [NginxProxyManager/nginx-proxy-manager#1527] [Q&A] [answered]
Hi,
As I was experiencing Bad Gateway problems (everything working well and suddenly not anymore after an upgrade of NPM), I saw that switching from MariaDB to SqlLite  could solve the problem
I switched, but now it seems NPM has begun all clean once again. I can see all the conf files for all the proxies I had previously setup, I can see the Certificate... 

Now I would like to know if it would be possible to get all my proxies and certificate just as they were before ?
How could I achieve that ?

Thanks in advance.

**Comment by chaptergy** [comment, marked as answer]
Unfortunately there is no easy way to do that, as an import / export function does not exist (https://github.com/jc21/nginx-proxy-manager/issues/168). The only thing you could do requires some knowlege about databases and a database tool like [DBeaver](https://dbeaver.io/). 

I have made a detailed write-up on how to do this in https://github.com/jc21/nginx-proxy-manager/discussions/1529#migrate-mariadb-to-sqlite-dbeaver

**Reply by mora-phi** [reply]
Hi, I almost did it, but it seems there is still a pending problem remaining.

First, some of the problems I found on my way :
- I was not able to directly copy data from one DB to the other, as it seemed something was happening but nothing was copied, so I resorted to exporting SQL Insert statements and executing them on the destination SQLite DB, which was not without problems because I had to truncate the rest before, etc... I then made sure all data was exactly spot on between origin DB and destination DB.
- Then, I copied the DB back, and to my surprise NPM would always create a new empty one named mydb.sqlite which seems to be used instead of database.sqlite... So I copied database.sqlite over mydb.sqlite
- After having two identical DBs, I would then encounter "migrations" errors in NPM's logs, so I had to remove all content of tables migration_lock and knex_migrations_lock.
- Next startup, NPM told me it could not find my certificate. For a reason, mine was in a folder named npm-2, and NPM would search in npm-2. Easy, I renamed all folders where it happened.

All in all I finally could get back into NPM without the Bad Gateway error, and all seems almost good, even though I now have this error when I try to display the list of hosts :
![image](https://user-images.githubusercontent.com/8950440/139080704-203ef48d-5dfe-4416-b9cb-dcc32b3d66fa.png)

[EDIT] Just checked in Chrome and I see this error in the Console :
![image](https://user-images.githubusercontent.com/8950440/139081823-6eea16be-e216-4135-8530-3a01e080a0f9.png)


Any idea what it could be ?

**Reply by chaptergy** [reply]
It seems something went wrong when you copied the data over. The `domain_names` column of the proxy hosts has to be a valid JSON array like `["npm.localhost"]`. If the brackets are missing or it is just a string, that could be the reason for this error.

**Reply by mora-phi** [reply]
Thanks a million, indeed I f#@ed up on of the domains in the DB when manually editing the SQL Inserts, because in MariaDB there is a "npm." prefix for each tablespace/table name that is not in SQLite (I guess that's the reason DBeaver did not auto-copya data), and I did a "replace all" that removed an important part of one of the domains. 
I was able to see which one debugging in Chrome in the loop of the error.

Nice, I now have it working again, many thanks !
